typescript/core/src/McapWriter.ts

```diff
@@ -469,4 +533,97 @@ export class McapWriter {
     await this.#writable.write(this.#recordWriter.buffer);
     this.#recordWriter.reset();
   }
+
+  /**
+   * Add attachment index from existing MCAP file.
+   * The purpose of this is to update the information for the new summary with the existing data from the existing MCAP file.
+   */
+  async #addAttachmentIndex(attachmentIndex: AttachmentIndex): Promise<void> {
+    if (this.statistics) {
+      ++this.statistics.attachmentCount;
+    }
+
+    if (this.#attachmentIndices) {
+      this.#attachmentIndices.push(attachmentIndex);
+    }
+
+    this.#dataSectionCrc = crc32Update(this.#dataSectionCrc, this.#recordWriter.buffer);
```

Comment:
I haven't dug in too closely but I suspect something is wrong with the CRC calculation, because I don't see the CRC being initialized in append mode.

It looks like _some_ of our readers (but not all) will verify this value when reading a mcap file (two examples I found are [python](https://github.com/foxglove/mcap/blob/main/python/mcap/mcap/stream_reader.py#L126) & [rust](https://github.com/foxglove/mcap/blob/main/rust/src/read.rs#L684)) – so you could lean on those libraries to validate the resulting crc is correct. I'm not sure if `mcap doctor` checks it.

You can also refer to the Go implementation: https://github.com/foxglove/mcap/blob/main/go/cli/mcap/utils/mcap_amendment.go

Reply:
I added CRC initialization for append mode, but am not quite sure what to set it to. Do you have a quick answer? If not, I can continue to look into the Go implementation and figure it out.